src/metabase/api/collection.clj

```diff
@@ -60,8 +60,11 @@
         (cons (root-collection namespace) collections))
       (hydrate collections :can_write)
       ;; remove the :metabase.models.collection.root/is-root? tag since FE doesn't need it
+      ;; and for personal collections we translate the name to user's locale
       (for [collection collections]
-        (dissoc collection ::collection.root/is-root?)))))
+        (-> collection
+            (dissoc ::collection.root/is-root?)
+            collection/personal-collection-with-ui-details)))))
```

Comment:
I actually put this logic in the `post-select` hook at first, but then I changed it because:
- IMO, hooks work best if we use them consistently, but from my reading, we have mixed use of toucan models and honeysql. You can see that for `GET /api/collection` we use toucan, but for `GET /api/collection/root/items` we use honeysql heavily.
- `post-select` has a big effect on other parts of the codebase and I'm not sure if we need the name translated to the user's locale in all cases.
- display name in the user's locale is a UI-specific need, so I think it'd be better if we make it clear in the code where we want it to happen - in this case it is at the API level.

That is my reasoning, let me know if you think it's still a good use of `post-select` for this case

Reply:
Fair points 👍